### PR TITLE
mdsameemali-adb-4nov

### PR DESCRIPTION
**Pull request by Sameem25**
This scrapes ADB Tenders data from website.

### CODE REVIEW

**Comment by gnommer** [comment]
1. there are no python class standards and just scripts mashed up in the repo
2. there is no sample output to understand the data.
3. the remaining modules are also missing in the repo

please address these points so we can consider the submission as valid

**Comment by KayvanShah1** [comment]
Please rename the PULL Request appropriately referring to submission guidelines. If you fail to do so we won't consider your assignment for further evaluation.

**Comment by Sameem25** [comment]
The changes are submitted. Please review.

File: dummy-data-product/src/dependencies/scraping/scraper.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+import dateutil
+import pandas as pd
+import numpy as np
+import pycountry
+import math
+import warnings
+warnings.filterwarnings('ignore')
+
+import sys
+sys.path.insert(1, 'C:/Users/Sam/Desktop/pt-mesh-pipeline-main/pt-mesh-pipeline-main/dummy-data-product/src/dependencies/cleaning')
```
Comment (KayvanShah1) [inline dummy-data-product/src/dependencies/scraping/scraper.py:15]: Bad practice of using hard-coded paths.

File: dummy-data-product/src/dependencies/scraping/scraper.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+sys.path.insert(1, 'C:/Users/Sam/Desktop/pt-mesh-pipeline-main/pt-mesh-pipeline-main/dummy-data-product/src/dependencies/standardization')
+import cleaning
+import standardizer
+
+#initiatin global variables
```
Comment (KayvanShah1) [inline dummy-data-product/src/dependencies/scraping/scraper.py:20]: Variables littered out the class

File: dummy-data-product/src/dependencies/scraping/scraper.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      #Since, map co-ordinates aren't given, puttin [] in place.
+      data['map_cordinates'] = '[]'
+      data['project_or_tender'] = 'T'   #scraping only tender data for this project
+      data['url'] = li7
+      data['source'] = li8
+      data['budget'] = li9
+      
+    return data
+
+  def clean(self):
```
Comment (KayvanShah1) [inline dummy-data-product/src/dependencies/scraping/scraper.py:137]: This is not meant to be implemented here.
It is a separate block after all.

File: dummy-data-product/src/dependencies/scraping/scraper.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+    data_to_clean = self.scrape()
+    clean_data = cleaning.cleaning(data_to_clean).clean()
+
+    return (clean_data)    
+
+  def standards(self):
```
Comment (KayvanShah1) [inline dummy-data-product/src/dependencies/scraping/scraper.py:144]: This is not meant to be implemented here.
It is a separate block after all.

File: dummy-data-product/src/dependencies/scraping/scraper.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    data_to_standardize = self.clean()
+    standardize_data = standardizer.standardize(data_to_standardize).standard()
+    return (standardize_data)
+  
+  def run(self):
+    output = self.standards()
+    return (output.to_csv('adb.csv', index = False))
+
+if __name__ == "__main__":
+  config = {}
```
Comment (KayvanShah1) [inline dummy-data-product/src/dependencies/scraping/scraper.py:155]: Why is this config empty?

How are you referring to the data you have scraped during the test run?
And how are you updating it?

**Review by KayvanShah1 — COMMENTED** [review]
Feedback:
- Poor data management
- Scripts are not clean
- Given boilerplate for writing python classes isn't followed(not improved upon after 1st review)
- Data from intermediate steps isn't saved.
- Metadata is absent
- Final data is incomplete
image
When you visit the URL for a particular tender, there are many important details available that are overlooked.

Final Word:
- The submitted assignment is not up to the mark to be considered for further process.

**Pull request closed without merging**